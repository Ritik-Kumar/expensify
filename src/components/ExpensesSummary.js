import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import setExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  return (
    <div>
      {(expensesCount == 1) ? (
        `Viewing ${expensesCount} expense totalling ${numeral(expensesTotal).format('$0,0.00')}`
      ) : (
          `Viewing ${expensesCount} expenses totalling ${numeral(expensesTotal).format('$0,0.00')}`
        )}
    </div>
  )
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: setExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);