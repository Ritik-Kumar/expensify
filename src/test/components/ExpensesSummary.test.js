import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expensesTotal from '../../selectors/expenses-total';

test('1 item', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={500} />);
  expect(wrapper).toMatchSnapshot();
});


test('2 item', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={500} />);
  expect(wrapper).toMatchSnapshot();
});