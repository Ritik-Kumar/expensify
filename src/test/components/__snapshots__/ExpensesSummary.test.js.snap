// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 item 1`] = `
<div>
  Viewing 1 expense totalling $500.00
</div>
`;

exports[`2 item 1`] = `
<div>
  Viewing 2 expenses totalling $500.00
</div>
`;
